bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -334,6 +336,18 @@ private boolean isBagsUpdate(final ISPObsComponent oldObsComp, final TargetObsCo
             return false;
         }
 
+        // REL-2822
+        // Okay if the scheduling block changed then it isn't just a bags update.
+        // Of course we don't know whether the scheduling block changed because
+        // that is kept in the observation node itself.  We can however look at
+        // the ephemeris for each nonsidereal target.
+        // Sorry, this is a terrible hack to make bulk scheduling block updates
+        // be considered enough of a change to refresh the target environment
+        // editor when it is selected..
+        if (!getNonSiderealTargets(oldTOC).equals(getNonSiderealTargets(newTOC))) {
+            return false;
+        }
```

Comment:
This is the worst part of the PR.  The issue was that the OT would be fooled into not updating the target editor for the last target component that was selected.  It would show the old guide star which is likely no longer workable with the new coordinates after a scheduling block change.  Fixing it required clicking on some other target node and then coming back to the same node, which was confusing.